lib/MultiTopicsConsumerImpl.cc

```diff
@@ -841,44 +823,47 @@ void MultiTopicsConsumerImpl::getBrokerConsumerStatsAsync(BrokerConsumerStatsCal
         callback(ResultConsumerNotInitialized, BrokerConsumerStats());
         return;
     }
+
     Lock lock(mutex_);
     MultiTopicsBrokerConsumerStatsPtr statsPtr =
         std::make_shared<MultiTopicsBrokerConsumerStatsImpl>(numberTopicPartitions_->load());
-    LatchPtr latchPtr = std::make_shared<Latch>(numberTopicPartitions_->load());
     lock.unlock();
-
     size_t i = 0;
-    consumers_.forEachValue([this, &latchPtr, &statsPtr, &i, callback](const ConsumerImplPtr& consumer) {
-        size_t index = i++;
-        auto weakSelf = weak_from_this();
-        consumer->getBrokerConsumerStatsAsync(
-            [this, weakSelf, latchPtr, statsPtr, index, callback](Result result, BrokerConsumerStats stats) {
+    // TODO: fix the thread safety issue if numberTopicPartitions_ was changed here
+    consumers_.forEachValue(
```

Comment:
Oh it's right. The previous code also has the same issue.

Reply:
I will revert the changes here. Using the new `forEachValue` API here will cause thread safety issue because the count of `SharedFuture` is the size of `consumers_` while  the count of `statsPtr` is `numberTopicPartitions_`.